lib/auth/httpfallback.go

```diff
@@ -631,24 +632,42 @@ func (c *Client) GetNodes(ctx context.Context, namespace string, opts ...service
 
 // GetAuthPreference gets cluster auth preference.
 func (c *Client) GetAuthPreference(ctx context.Context) (types.AuthPreference, error) {
-	if resp, err := c.APIClient.GetAuthPreference(ctx); err != nil {
+	authPref, err := c.APIClient.GetAuthPreference(ctx)
+	if err != nil {
 		if !trace.IsNotImplemented(err) {
 			return nil, trace.Wrap(err)
 		}
-	} else {
-		return resp, nil
-	}
-	out, err := c.Get(c.Endpoint("authentication", "preference"), url.Values{})
-	if err != nil {
-		return nil, trace.Wrap(err)
+		out, err := c.Get(c.Endpoint("authentication", "preference"), url.Values{})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		authPref, err = services.UnmarshalAuthPreference(out.Bytes())
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
 	}
 
-	cap, err := services.UnmarshalAuthPreference(out.Bytes())
+	// AuthPreference was updated in 7.0.0 to hold legacy cluster config fields. If the
+	// server version is < 7.0.0, we must update the AuthPreference with the legacy fields.
+	minServerVersion := "7.0.0"
```

Comment:
FWIW, this will also catch versions like `7.0.0-beta.1` (that already implement the new endpoints) since they are `< 7.0.0` according to the SemVer spec. You might consider moving this workaround https://github.com/gravitational/teleport/blob/b4006bcfc9f30e60def4cd2bb4496a4ba9da9fb4/lib/reversetunnel/srv.go#L1081 to `lib/utils/ver.go` and then using it here as well.

Reply:
Instead I added `utils.SetVersionBeforeAlpha` and went with the strategy of appending `"-aa"` to the version. This is being reused in a few places and won't need to be deleted.